test/v0.8/testcases/module/matching/PublishMatchingTestSuite.sol

```diff
@@ -82,5 +276,26 @@ contract PublishMatchingTestCaseWithSuccess is MatchingsTestBase {
             100,
             "TEST"
         );
+        return datasetId;
+    }
+
+    function action(uint64 _datasetId) internal virtual override {
+        uint64 carsCount = matchings.datasets().getDatasetCarsCount(
+            _datasetId,
+            DatasetType.DataType.MappingFiles
+        );
+        bytes32[] memory cars = new bytes32[](carsCount);
```

Comment:
No need to allocate memory here. If you need to copy the data value to new memory, you cannot use it like this.